### DISCUSSION TITLE
Unable to upload to STM32

### DISCUSSION THREAD

**Discussion by TTXL3099** [stm32duino/Arduino_Core_STM32#2393] [Q&A] [answered]
**Describe the bug**
Unable to upload using maple dfu for STM32F103C8T6

**To Reproduce**
I used MAPLE DFU2.0 and burned the BootLoader through stm32prg using ST-LINK. When uploading for the first time, the output is like this.But HID BootLoader works well.

It seems that the upload failed, but the program ran correctly. When I plugged in and unplugged the STM32 again, the device manager prompted as follows. It seems that the BootLoader was overwritten?I made sure I selected USB-CDC.I made sure I followed every step according to [https://github.com/stm32duino/Arduino_Core_STM32/wiki/Upload-methods](https://github.com/stm32duino/Arduino_Core_STM32/wiki/Upload-methods) and [https://www.instructables.com/Programming-STM32-BluePill-Via-Arduino-for-Beginne/](https://www.instructables.com/Programming-STM32-BluePill-Via-Arduino-for-Beginne/)

**Expected behavior**
Use maple bootloader to upload the program.

**Screenshots**
![屏幕截图 2024-06-07 141215](https://github.com/stm32duino/Arduino_Core_STM32/assets/95993288/d6a73273-ac65-45c1-b6f6-dc2e055424a8)
![屏幕截图 2024-06-07 141556](https://github.com/stm32duino/Arduino_Core_STM32/assets/95993288/50c8efe9-d434-4ae3-b5cc-9b6cb9e7f9a4)
![image](https://github.com/stm32duino/Arduino_Core_STM32/assets/95993288/7bb91996-c153-420c-8f08-b83aa43cffd1)
**Desktop (please complete the following information):**
 - Windows10 Pro 22H2
 - Arduino IDE version: 2.3.2
 - STM32 core version: 2.7.1
 - Tools menu settings if not the default
 - Upload method: Maple DFU BootLoader 2.0

**Board (please complete the following information):**
 - Name: STM32F103C8T6(made in PRC) 
![image](https://github.com/stm32duino/Arduino_Core_STM32/assets/95993288/40376f0a-c481-492a-8325-5e52c9c6984c)
![image](https://github.com/stm32duino/Arduino_Core_STM32/assets/95993288/1943677f-0503-475f-8ffb-eaeb58835806)
 - Hardware Revision: NONE
 - Extra hardware used if any: NONE

**Additional context**

**Comment by fpistm** [comment]
The error is not linked to the upload but the reset after the upload, it is a know issue with dfu.
I see no issue here, and tested a lot the upload recently and no issue. Why you tell it is overwritten ? The flash use the right offset so I don't see any issue.

**Reply by TTXL3099** [reply]
When I replug and plug the STM32 after uploading, the computer prompts me "Device Descriptor Request Failed" (-43), just like the second picture. Is this due to a configuration problem?Thanks.

**Reply by TTXL3099** [reply]
How should I proceed with Maple BootLoader after uploading it?

**Reply by TTXL3099** [reply]
Judging from the beeps from Windows, it seems that when I plug the STM32 in again, a device is connected - removed - connected again - and then reports that the device descriptor request failed.

**Reply by fpistm** [reply]
USB enumeration issue with BluePill is know issue. Look at the forum about a resistor issue.

**Reply by TTXL3099** [reply]
Is there a specific web link? In the second web link I mentioned there is a reference to the resistance issue with the R10-PA12. Apparently my STM32 is normal - it says 152 on the R10.

**Comment by fpistm** [comment, marked as answer]
Select bluepill f193c8 in the part number menu

**Reply by TTXL3099** [reply]
Amazing!!
![image](https://github.com/stm32duino/Arduino_Core_STM32/assets/95993288/a92cf4d0-f2bb-4ad7-b141-81bad01a1fdf)
![image](https://github.com/stm32duino/Arduino_Core_STM32/assets/95993288/e6427479-57a7-42db-89da-ee4f6bd21b80)
Can you tell me why it works?

**Reply by fpistm** [reply]
Probably due to clock config. Generic uses HSI and it is probably not enough precise 
.

**Reply by TTXL3099** [reply]
Thank you very much for helping me solve the problem!On the occasion of the Dragon Boat Festival, a traditional Chinese festival,I wish you a peaceful and healthy Dragon Boat Festival!
